actionmailer/lib/action_mailer/base.rb

```diff
@@ -973,10 +973,11 @@ def collect_responses_from_templates(headers)
         templates_name = headers[:template_name] || action_name
 
         each_template(Array(templates_path), templates_name).map do |template|
-          self.formats = [template.format]
+          format = template.format || self.formats.first
+          self.formats = [format]
           {
             body: render(template: template),
-            content_type: template.type.to_s
+            content_type: Mime[format].to_s
```

Comment:
Why using `Mime` here and not `Types` like in the other places?

Reply:
I debated which to use here. This is inside actionmailer which [makes `Types` delegate to `Mime`](`https://github.com/rails/rails/blob/master/actionmailer/lib/action_mailer.rb#L67-L70`), so they should be the same and I felt it made more sense to just use `Mime` than `ActionView::Template::Types` here. I'm happy to switch it if `ActionView::Template::Types` would be better!

Reply:
oh. I didn't see it was inside Action Mailer. So Mime is better here indeed.